### PR TITLE
`Integration Tests`: added test for `promotionalOffer`

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

File: ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift [hunk 1]
```diff
@@ -116,6 +116,34 @@ class StoreKit1IntegrationTests: BaseIntegrationTests {
...
+        }
+    }
+
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, *)
+    func testPromotionalOffer() async throws {
+        try await self.purchaseMonthlyOffering()
+        try self.testSession.expireSubscription(productIdentifier: Self.productIdentifier)
+
+        var result = try await CommonFunctionality.promotionalOffer(
+            for: Self.productIdentifier,
```
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift:139]: We could also add a test that checks that it fails getting a promotional offer if passing an incorrect `productIdentifier` or `discountIdentifier`

**Review by tonidero — APPROVED** [review]
❤️ Love this! I wish we had such test infrastructure on Android...

File: ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift (hunk 1 above)
Comment (tonidero) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift:139]: Thinking about it more, it might be too much to try to cover all edge cases in integration tests, so I think it's ok to cover only the major cases like this.

Comment (NachoSoto) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift:139]: It doesn't hurt though, it's trivial. Just added it.
